main.py

```diff
@@ -117,36 +118,24 @@ def run(filename, rule_type=RuleType.ALL, with_console_output=False, execution_m
                 version = len(shas)
-                check_disabled = 'disabled' in item['tags']
-                if check_disabled:
-                    yield f"{feature_name}.v{version}", f"{remote}/blob/{shas[0]}/{feature_file}", "Rule disabled", ("Rule disabled", "This rule has been disabled from checking"), "Rule disabled", "Rule disabled"
-
+                rule_is_disabled = 'disabled' in item['tags']
+                yield { # add feature information, mainly to check disabled feature
+                    'feature_name': f"{feature_name}.v{version}",
+                    'remote' : f"{remote}/blob/{shas[0]}/{feature_file}",
+                    'rule_is_disabled': rule_is_disabled
+                }
                 try:
                     el_list = item['elements']
                 except KeyError:
                     el_list = []
                 for el in el_list:
-                    scenario_name = el['name']
-                    for step in el['steps']:
-                        step_name = step['name']
-                        step_status = step.get('result', {}).get('status')
-                        instance_id = "-"
-                        if step_status and step['step_type'] == 'then':
-                            try:
-                                results = step['result']['error_message']
-                            except KeyError:  # THEN not checked
-                                results = []
-                            except json.decoder.JSONDecodeError:  # THEN not checked
-                                results = []
-                            if results:
-                                pattern = r'ifc_instance_id=(\d+)'
-                                match = re.search(r'ifc_instance_id=(\d+)', str(results))
-                                if match:
-                                    instance_id = '#' + match.group(1)
-                                    results = re.sub(pattern, '', str(results))
-                                yield f"{feature_name}/{scenario_name}.v{version}", f"{remote}/blob/{shas[0]}/{feature_file}", f"{scenario_name}", f"{step_name}", instance_id, results
-                            elif item['activated']:
-                                yield f"{feature_name}/{scenario_name}.v{version}", f"{remote}/blob/{shas[0]}/{feature_file}", f"{scenario_name}" , f"{step_name}", instance_id, "Rule passed"
+                    scenario_validation_outcomes = json.loads(base64.b64decode(el.get('validation_outcomes', [{}])[0].get('data', '')).decode('utf-8')) if el.get('validation_outcomes') else []
+                    for validation_outcome in scenario_validation_outcomes:
+                        validation_outcome.update({ # add scenario information to display in terminal
+                            'scenario_name': el['name'],
+                            'step_names': [step['name'] for step in el['steps']]
+                        })
+                        yield validation_outcome
```

Comment:
I think it's a little bit nicer to move this out of the loop.


```suggestion
                    scenario_info = {
                            'scenario_name': el['name'],
                            'step_names': [step['name'] for step in el['steps']]
                        }
                    for validation_outcome in scenario_validation_outcomes:
                        yield validation_outcome | scenario_info
```

(indentation needs to be fixed)

Reply:
[That's nicer indeed, thanks](https://github.com/buildingSMART/ifc-gherkin-rules/commit/101ec619eb8815e7f074479aaf4199ab1a8f3800).